### ISSUE TITLE
QA Feedback on lesson

### ISSUE THREAD

**Issue by KatieMFritz** [KatieMFritz/codelab-js-arrays#1]
@chrisvfritz
## Managing lists of items with arrays

Do you want to remind people they can follow along in their browser console?

---

> The reason an index starts at 0, is it describes the distance from the start of the array. The first element is at the start of the array, so there is no distance. It's 0.

This would be a good place for a simple number line diagram, to really bring this point home. I think I saw one recently and that's part of why I understand this already.

---

Great step-by-step, practical example.

**Comment by KatieMFritz** [comment]
## Adding items to an array

> `splice.` Yeah, it has another weird name. 

Splice makes sense to me actually - you're chopping the array into two pieces, then splicing them together with the item you add. Oh. You're way ahead of me. PS DON'T GOOGLE IMAGE SEARCH SPLICE.

If you want to add a more down to earth example, http://www.wikihow.com/Splice-Wire.

**Comment by KatieMFritz** [comment]
## Removing items from arrays

> books.splice(1, 1)
> => [3]

You mean `numbers.splice`

**Comment by KatieMFritz** [comment]
## Creating a todo list

> Now let's use our mad array modification skills

I enjoy the phrase mad skills, but with the line break it's a little hard to parse - I caught myself thinking "what's a mad array?." How about "newfound... skills" or just "skills"?

---

> Do you see that join('') on the last line below? 

above, not below

---

Sweet, just made an app.

**Comment by KatieMFritz** [comment]
## Removing items from our todo list

> We can't specific the click behavior for elements that don't even exist yet! 

specify

---

> Instead of watching for an event on the elements that will be dynamically added and removed, you can watch for the event on the containing element that will always be there. In this case, that's our todoList element.

I had to read this three times to get it. It might just be because I'm tired though. I'll check back tomorrow. Maybe incorporate the examples into the sentence, instead of after?

Instead of watching for an event on the elements that will be dynamically added and removed **(our list items)**, you can watch for the event on the _containing element_ that will always be there **(our `todoList` element)**. ~~In this case, that's our todoList element.~~

---

> Now we only have one problem remaining. When a button is clicked, how will we know _which_ todo to remove.

Now we only have one problem remaining**:** When a button is clicked, how will we know _which_ todo to remove**?**

---

> In this case, we want to add the index of the item, which we can get from a second parameter in our map function, then set it to a data-index attribute.
> Now we can access the dataset.index of our clicked element to get the index of the item we should remove from the array, then call our renderTodos function to update the page.

This moved a little fast, but by reading it a few times and carefully studying the code, I think I get it. Keep an eye out for students getting stuck here, and if they do, break it down further.

**Comment by chrisvfritz** [comment]
@KatieMFritz Could you show me an example of a line diagram? I'm not exactly sure what you mean by it.

**Comment by KatieMFritz** [comment]
@chrisvfritz Like this, more or less: 
<img src="http://etc.usf.edu/clipart/41600/41654/0-10_41654_lg.gif">

I'll sketch one for you on paper.

**Comment by KatieMFritz** [comment]
## Now it's your turn

> If you run into a problem you're not sure how to solve, don't hesitate Googling for more information.

If you run into a problem you're not sure how to solve, don't hesitate **to Google** ~~Googling~~ for more information.

**Comment by chrisvfritz** [comment]
OK, I had something like this to try to make that clearer:

![](https://www.dropbox.com/s/0q4n9ucpdbo1okj/Screenshot%202016-01-27%2011.02.25.png?dl=1)

Do you think I should try to add a line down the left side?
